nilearn/plotting/surface/_matplotlib_backend.py

```diff
@@ -319,429 +320,447 @@ def _threshold_and_rescale(data, threshold, vmin, vmax):
-                curv_map = self.load_surf_data(surf_mesh[f"curv_{hemi}"])
-                curv_sign_map = (np.sign(curv_map) + 1) / 4 + 0.25
-                bg_map = curv_sign_map
-            else:
-                sulc_map = surf_mesh[f"sulc_{hemi}"]
-                bg_map = sulc_map
-
-            ax = fig.add_subplot(grid[i + len(hemis)], projection="3d")
-            axes.append(ax)
-
-            self.plot_surf_stat_map(
-                surf[hemi],
-                texture[hemi],
-                view=mode,
-                hemi=hemi,
-                bg_map=bg_map,
-                bg_on_data=bg_on_data,
-                axes=ax,
-                colorbar=False,  # Colorbar created externally.
-                vmin=vmin,
-                vmax=vmax,
-                threshold=threshold,
-                cmap=cmap,
-                symmetric_cbar=symmetric_cbar,
-                **kwargs,
-            )
 
-            # We increase this value to better position the camera of the
-            # 3D projection plot. The default value makes meshes look too
-            # small.
-            ax.set_box_aspect(None, zoom=1.3)
+        cbar_grid = GridSpecFromSubplotSpec(3, 3, grid[-1, :])
+        cbar_ax = fig.add_subplot(cbar_grid[1])
+        axes.append(cbar_ax)
+        # Get custom ticks to set in colorbar
+        ticks = _get_ticks(vmin, vmax, cbar_tick_format, threshold)
+        fig.colorbar(
+            sm,
+            cax=cbar_ax,
+            orientation="horizontal",
+            ticks=ticks,
+            format=cbar_tick_format,
+        )
 
-        if colorbar:
-            sm = _colorbar_from_array(
-                self.get_image_data(stat_map),
-                vmin,
-                vmax,
-                threshold,
-                symmetric_cbar=symmetric_cbar,
-                cmap=plt.get_cmap(cmap),
-            )
+    if title is not None:
+        fig.suptitle(title, y=1.0 - title_h / sum(height_ratios), va="bottom")
 
-            cbar_grid = GridSpecFromSubplotSpec(3, 3, grid[-1, :])
-            cbar_ax = fig.add_subplot(cbar_grid[1])
-            axes.append(cbar_ax)
-            # Get custom ticks to set in colorbar
-            ticks = _get_ticks(vmin, vmax, cbar_tick_format, threshold)
-            fig.colorbar(
-                sm,
-                cax=cbar_ax,
-                orientation="horizontal",
-                ticks=ticks,
-                format=cbar_tick_format,
-            )
+    if output_file is None:
+        return fig, axes
+    fig.savefig(output_file, bbox_inches="tight")
+    plt.close(fig)
 
-        if title is not None:
-            fig.suptitle(
-                title, y=1.0 - title_h / sum(height_ratios), va="bottom"
-            )
 
-        if output_file is None:
-            return fig, axes
-        fig.savefig(output_file, bbox_inches="tight")
-        plt.close(fig)
+def adjust_colorbar_and_data_ranges(
+    stat_map, vmin=None, vmax=None, symmetric_cbar=None
+):
```

Comment:
For private functions, I still like to have a one-line docstring saying what it does.